maven/flatpak-maven-generator

```diff
@@ -0,0 +1,222 @@
+    OKCYAN = '\033[96m'
+    OKGREEN = '\033[92m'
+    WARNING = '\033[93m'
+    FAIL = '\033[91m'
+    ENDC = '\033[0m'
+
+
+def assembleUri(repo: str, groupId: str, artifactId: str, version: str, extension: str) -> str:
+    groupId = groupId.replace(".", "/")
+    return f'{repo}{groupId}/{artifactId}/{version}/{artifactId}-{version}.{extension}'
+
+def getFileHash(file) -> str:
+    file.seek(0)
+    bytes = file.read() # read entire file as bytes
+    return hashlib.sha256(bytes).hexdigest();
+
+def parsePomDeps(parsed_pom) -> list[object]:
+    ns = {'POM': 'http://maven.apache.org/POM/4.0.0'}
+
+    result = []
+
+    parent = parsed_pom.find("POM:parent", ns)
+    if parent is not None:
+        groupId = parent.find("POM:groupId", ns).text
+        artifactId = parent.find("POM:artifactId", ns).text
+        version = parent.find("POM:version", ns).text
+
+        result.append({
+            "groupId": groupId,
+            "artifactId": artifactId,
+            "version": version
+        })
+
+    deps = parsed_pom.findall("POM:dependencies/POM:dependency", ns)
+    for dep in deps:
+        groupId = dep.find("POM:groupId", ns).text
+        artifactId = dep.find("POM:artifactId", ns).text
+        version = dep.find("POM:version", ns)
+
+        if(version is None):
+            continue
+
+        version = version.text
+
+        result.append({
+            "groupId": groupId,
+            "artifactId": artifactId,
+            "version": version
+        })
+
+    return result
+
+def getPackagingType(parsed_pom) -> str:
+    ns = {'POM': 'http://maven.apache.org/POM/4.0.0'}
+
+    packaging = parsed_pom.find("POM:packaging", ns)
+    if packaging is not None and packaging.text == "pom":
+        return None # Nothing to download for this
+
+    if packaging is None:
+        # jar is default if nothing is specified
+        return "jar"
+
+    return packaging.text
+
+modules = []
+addedModules = []
+
+def addModule(groupId: str, artifactId: str, version: str):
+    addedModules.append({
+        "groupId": groupId,
+        "artifactId": artifactId,
+        "version": version
+    })
+
+def downloadAndAdd(repo: str, groupId: str, artifactId: str, version: str, binaryType: str) -> bool:
+    url = assembleUri(repo, groupId, artifactId, version, binaryType)
+    groupId = groupId.replace(".", "/")
+    try:
+        with urllib.request.urlopen(url) as response:
+            with tempfile.NamedTemporaryFile(delete=True) as tmp_file:
+                shutil.copyfileobj(response, tmp_file)
+
+                modules.append({
+                   "type": "file",
+                   "url": url,
+                   "sha256": getFileHash(tmp_file),
+                   "dest": f"maven-local/{groupId}/{artifactId}/{version}"
+                })
+
+                return True
+    except urllib.error.HTTPError as e:
+        print(f"{colors.WARNING}Unable to download {binaryType} file specified in metadata{colors.ENDC}")
```

Comment:
Consider using `logging` module instead. And if you want to keep colours, you can use a custom log formatter to add them.
The same applies to other debug prints.